### PR TITLE
Weekly Patch Release v1.3.7 [full merge, no squash]

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

See #7467

```fish
adrian@quantumgrid ~/r/pytorch-lightning (master)> gh pr list -s merged -S 'merged:2021-06-15T12:00:00.000Z..2021-06-22T23:30:00.000Z' --json mergedAt,milestone,url,mergeCommit,title --jq 'sort_by(.mergedAt) | reverse | .[] | select((.milestone.title == "v1.3.x") or (.milestone.title == null)) | [.url, .mergeCommit.oid, .title] | join(" ")' --limit 100

OK https://github.com/PyTorchLightning/pytorch-lightning/pull/8002 dd340a659818def94468f11aa4fed492e27f3e92 Actually show deprecation warnings and their line level [2/2]
OK https://github.com/PyTorchLightning/pytorch-lightning/pull/7959 2303f9ced81e8f9e1c8ec62aaa03a78e9e5efaa3 Fix(Early Stopping): move best score to device
OK https://github.com/PyTorchLightning/pytorch-lightning/pull/8009 651a0fbdeb97a6f73e6b87af4eabd1bd53a9c065 [bugfix] Properly name PyTorchProfiler traces
OK https://github.com/PyTorchLightning/pytorch-lightning/pull/7814 e808f9fb28aba9b356109b038f5517c4a0581ce9 Use DistributedSampler when running with custom accelerator
OK https://github.com/PyTorchLightning/pytorch-lightning/pull/8005 f447839d16f1d7861291c76aae1945cc62d3455e Add `warning_cache.deprecation` and set warning stacklevel [1/2]
OK https://github.com/PyTorchLightning/pytorch-lightning/pull/7841 55494e87452a7521ecd2fb161fe7a38ae5ee20e4 Fix Special Tests
-- https://github.com/PyTorchLightning/pytorch-lightning/pull/7997 917cf8363808095a6699c0adb15669af44f69140 [doc] Add more reference around predict_step
OK https://github.com/PyTorchLightning/pytorch-lightning/pull/7975 b71aa55b9e81dd5cabb0ddf6b57dae916de6f2bf Make optimizers skippable when using amp


```

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by kaushikb11 — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
LGMT !

File: pytorch_lightning/core/lightning.py [hunk 1]
```diff
@@ -165,10 +165,6 @@ def example_input_array(self, example: Any) -> None:
 
     @property
     def datamodule(self) -> Any:
-        rank_zero_deprecation(
```
Comment (carmocca) [inline pytorch_lightning/core/lightning.py:168]: Shouldnt be gone

File: tests/deprecated_api/test_remove_1-5.py [hunk 2]
```diff
@@ -362,15 +362,6 @@ def test_v1_5_0_trainer_gpus_str_parsing(*_):
         assert gpus == [0]
 
 
-def test_v1_5_0_datamodule_setter():
```
Comment (carmocca) [inline tests/deprecated_api/test_remove_1-5.py:365]: Same

Comment (awaelchli) [reply tests/deprecated_api/test_remove_1-5.py]: added back!

File: CHANGELOG.md [hunk 3]
```diff
@@ -4,6 +4,13 @@ All notable changes to this project will be documented in this file.
 
 The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 
+## [1.3.7] - 2021-06-22
+
+- Fixed a bug where skipping an optimizer while using amp causes amp to trigger an assertion error ([#7975](https://github.com/PyTorchLightning/pytorch-lightning/pull/7975))
```
Comment (carmocca) [inline CHANGELOG.md:9]:
```suggestion

- Fixed a bug where skipping an optimizer while using amp causes amp to trigger an assertion error ([#7975](https://github.com/PyTorchLightning/pytorch-lightning/pull/7975))
```

**Review by carmocca — APPROVED** [review]

**Review by SkafteNicki — APPROVED** [review]

File: .azure-pipelines/ipu-tests.yml [hunk 4]
```diff
@@ -53,12 +53,9 @@ jobs:
         export GIT_TERMINAL_PROMPT=1
         python -c "fname = 'requirements/extra.txt' ; lines = [line for line in open(fname).readlines() if 'fairscale' not in line] ; open(fname, 'w').writelines(lines)"
         python -c "fname = 'requirements/extra.txt' ; lines = [line for line in open(fname).readlines() if 'horovod' not in line] ; open(fname, 'w').writelines(lines)"
-
         python ./requirements/adjust_versions.py requirements/extra.txt
         python ./requirements/adjust_versions.py requirements/examples.txt
-
-        pip install --requirement ./requirements/devel.txt --upgrade-strategy only-if-needed
-
+        pip install . --requirement requirements/devel.txt
```
Comment (Borda) [inline .azure-pipelines/ipu-tests.yml:58]: uninstall it at the end

File: CHANGELOG.md [hunk 5]
```diff
@@ -4,6 +4,14 @@ All notable changes to this project will be documented in this file.
 
 The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 
+## [1.3.7] - 2021-06-22
+
+
```
Comment (Borda) [inline CHANGELOG.md:9]:
```suggestion
```

**Review by Borda — CHANGES_REQUESTED** [review]

File: .azure-pipelines/ipu-tests.yml (hunk 4 above)
Comment (awaelchli) [reply .azure-pipelines/ipu-tests.yml:58]: i cherry picked this from master, as I was told we need ci updates in 1.3.x

**Review by Borda — APPROVED** [review]
my bad thought it is normal CI, not IPU...

**Pull request merged**